druid/src/widget/textbox.rs

```diff
@@ -365,13 +371,30 @@ impl Widget<String> for TextBox {
                 self.update_hscroll(&text_layout);
             }
         }
+
+        ctx.set_focus_node(previous_focus_node);
     }
 
     fn lifecycle(&mut self, ctx: &mut LifeCycleCtx, event: &LifeCycle, _data: &String, _env: &Env) {
         match event {
-            LifeCycle::WidgetAdded => ctx.register_for_focus(),
+            LifeCycle::WidgetAdded => {
+                let previous_focus_node = ctx.focus_node();
+                self.focus_node.widget_id = Some(ctx.widget_id());
+                ctx.set_focus_node(self.focus_node);
+                ctx.register_for_focus();
+                ctx.set_focus_node(previous_focus_node);
```

Comment:
it's a bit concerning that I would have to do this much bookkeeping in order to create a custom widget that participated in focus without using a `Focus` wrapper? It might make sense to only allow focus to work via `Focus`, if it's a significant simplification.

Reply:
Yes, definitely, only the `Focus` widget should manage focus; I want to do this in the next iteration. It will already work fine if delete all stuff with focus management in the `TextBox`, but you should always wrap this widget with `Focus,` and this will break backward compatibility.